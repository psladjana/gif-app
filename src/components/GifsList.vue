<template>
   <Carousel
    :autoplay="true"
    :autoplayTimeout="5000"
    :scrollPerPage="true"
    :navigationEnabled="true"
    :paginationEnabled="false"
    :loop="true"
    :perPageCustom="[[480, 1], [768, 3], [992, 4]]">
      <Slide
        v-for="gif in gifs"
        :key="gif.id">
        <GifListItem :gif="gif" :loading='loading'/>
      </Slide>
    </Carousel>
</template>

<script>
import { Carousel, Slide } from 'vue-carousel';
import GifListItem from '@/components/GifListItem.vue';

export default {
  name: 'GifList',
  components: {
    GifListItem,
    Carousel,
    Slide,
  },
  props: {
    gifs: Array,
    loading: Boolean,
  },
};
</script>
