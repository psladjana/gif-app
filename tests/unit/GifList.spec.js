import { shallowMount } from '@vue/test-utils';
import GifsList from '@/components/GifsList.vue';

describe('GifsList.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(GifsList);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders props.gifs when passed', () => {
    const gifs = [];
    const wrapper = shallowMount(GifsList, {
      propsData: { gifs },
    });
    expect(wrapper.vm.$props.gifs).toEqual(gifs);
  });
});
