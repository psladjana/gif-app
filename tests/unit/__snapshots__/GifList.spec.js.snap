// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifsList.vue matches snapshot 1`] = `<carousel-stub autoplay="true" autoplaytimeout="5000" autoplayhoverpause="true" autoplaydirection="forward" easing="ease" loop="true" minswipedistance="8" mousedrag="true" touchdrag="true" navigateto="0" navigationclicktargetsize="8" navigationenabled="true" navigationnextlabel="&amp;#9654" navigationprevlabel="&amp;#9664" paginationactivecolor="#000000" paginationcolor="#efefef" paginationpadding="10" paginationposition="bottom" paginationsize="10" perpage="2" perpagecustom="480,1,768,3,992,4" resistancecoef="20" scrollperpage="true" spacepadding="0" spacepaddingmaxoffsetfactor="0" speed="500" tagname="slide" maxpaginationdotcount="-1"></carousel-stub>`;
